src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -128,10 +132,17 @@ public function authenticate(RequestEvent $event)
         }
 
         if ($token instanceof TokenInterface) {
+            $originalToken = $token;
             $token = $this->refreshUser($token);
 
-            if (!$token && $this->rememberMeServices) {
-                $this->rememberMeServices->loginFail($request);
+            if (!$token) {
+                if (null !== $this->firewallDispatcher) {
+                    $this->firewallDispatcher->dispatch(new TokenDeauthenticatedEvent($originalToken, $request));
+                }
+
+                if ($this->rememberMeServices) {
+                    $this->rememberMeServices->loginFail($request);
+                }
```

Comment:
Is this now a deprecated path? Only the old system would use this?

Reply:
Yes